scripts/run_e2e_tests.py

```diff
@@ -320,20 +319,32 @@ def main(args=None):
                 break
 
             # Check whether we should rerun based on this suite's policy.
-            test_is_flaky = flake_checker.is_test_output_flaky(
+            test_is_flaky, rerun = flake_checker.is_test_output_flaky(
                 output, parsed_args.suite)
-            if policy == RERUN_POLICY_NEVER:
-                print(
-                    'Not rerunning because the policy is to never '
-                    'rerun the {} suite'.format(parsed_args.suite))
-                break
-            if policy == RERUN_POLICY_KNOWN_FLAKES and not test_is_flaky:
-                print((
-                    'Not rerunning because the policy is to only '
-                    'rerun the %s suite on known flakes, and this '
-                    'failure did not match any known flakes')
-                    % parsed_args.suite)
-                break
+            if rerun != flake_checker.RERUN_UNKNOWN:
+                if rerun == flake_checker.RERUN_YES:
+                    print(
+                        'Rerunning as per instructions from logging '
+                        'server.')
+                else:
+                    assert rerun == flake_checker.RERUN_NO
+                    print(
+                        'Not rerunning as per instructions from '
+                        'logging server.')
+                    break
+            else:
```

Comment:
instead of this nested if-else can we do something around:
```
if rerun=yes:
     ...
     continue

if rerun=no:
   ...
   break

if rerun!=unknown:
   ...
   break

if policy = ....
 ```

Reply:
done